patacrep/templates.py

```diff
@@ -84,11 +88,31 @@ def __init__(self, template, jinjaenv, encoding=None):
         self.jinjaenv.filters['escape_tex'] = _escape_tex
         self.jinjaenv.trim_blocks = True
         self.jinjaenv.lstrip_blocks = True
-        self.jinjaenv.filters["path2posix"] = files.path2posix
-        self.jinjaenv.filters["iter_datadirs"] = files.iter_datadirs
-        self.jinjaenv.filters["lang2babel"] = lang2babel
+        self._fill_filters()
         self.template = self.jinjaenv.get_template(template)
 
+    def _fill_filters(self):
+        """Define some jinja2 filters, if not set yet."""
+        for key, value in [
+                ("path2posix", files.path2posix),
+                ("iter_datadirs", files.iter_datadirs),
+                ("lang2babel", self.lang2babel),
```

Comment:
J'ai l'impression qu'on se mélange assez facilement les pinceaux avec la gestion des langues...

Une solution pourrait être de créer une classe `Language` et une `FallbackLanguage` (instanciations faites par [`checklanguage`](https://github.com/patacrep/patacrep/blob/errors/patacrep/latex/__init__.py#L109), qui pourrait devenir `get_language`):

```py
class Language:
    """Language."""

    def __init__(self, *, code):
        self.code = code

    @property
    def babel(self):
        """Return the fallback babel language."""
        return BABEL_LANGUAGES[self.code]

    def __str__(self):
        return self.code

class FallbackLanguage(Language):
    """Fallback language because the original code or babel lang is not supported."""

    def __init__(self, *, code, original_code=None, original_babel=None):
        self.code = code
        self.original_code = original_code
        self.original_babel = original_babel
```

Et lors de l'interprétation des chants, on regarde si la langue est `Language` ou `FallbackLanguage` pour générer l'erreur correspondante.